src/main/java/com/revo/deployr/client/util/RDataUtil.java

```diff
@@ -174,25 +174,35 @@ private static RData buildPrimitiveFromJSON(String name, Map json) {
         String rclass = (String) json.get("rclass");
 
         log.debug("buildPrimitiveFromJSON: name=" + name + " type=" + type + " rclass=" + rclass + " json=" + json);
-
+    
         if (json.containsKey("value")) {
 
             Object value = json.get("value");
 
             log.debug("buildPrimitiveFromJSON: name=" + name + " has value=" + value);
 
-            if (value instanceof Integer) {
-                rData = new RNumericImpl(name, ((Integer) value).doubleValue());
-            } else if (value instanceof Double) {
-                rData = new RNumericImpl(name, (Double) value);
-            } else if (value instanceof String) {
-                rData = new RStringImpl(name, (String) value);
-            } else if (value instanceof Boolean) {
-                rData = new RBooleanImpl(name, (Boolean) value);
-            } else {
-                // Indeterminate type, skip here to treat as RDataNA.
+            if("character".equalsIgnoreCase(rclass)){
+            	rData = new RStringImpl(name,  value == null ? null : (String) value);            	
+            }else if("numeric".equalsIgnoreCase(rclass)){
+            	rData = new RNumericImpl(name, value == null ? null : (Double) value);
+            }else if("logical".equalsIgnoreCase(rclass)){
+            	rData = new RBooleanImpl(name, value == null ? null :(Boolean) value);
             }
 
+            if(rData == null){
```

Comment:
Original logic is now inside the if(radata == null) check, new logic that was added it to use rClass type which we get as part of response and use that to determine the type of RData object that we will create.
we fall back to old logic only when we don't create rData object.